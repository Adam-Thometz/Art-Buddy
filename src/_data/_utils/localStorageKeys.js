// visited state for Word-To-Music Decoder
export const WTM = 'wtm';
// visited state for Jump Into Rhythm
export const JIR = 'jir';
// visited state for Sequence Maker
export const SM = 'sm';

// visited state for Free Paint
export const FP = 'fp';

// visited state for Score Keeper
export const SK = 'sk';
// visited state for Time Keeper
export const TK = 'tk';
// visited state for Mood Meter
export const MM = 'mm';