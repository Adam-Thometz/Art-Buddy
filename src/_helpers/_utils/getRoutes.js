/** getRoutes
 * Purpose: takes a React component and returns its children. Mainly used to get Route components found in a larger component and spread them in a Routes component.
 */

export default function getRoutes(component) {
  return component().props.children;
};