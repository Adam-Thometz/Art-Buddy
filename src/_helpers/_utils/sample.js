import { Sampler, Volume } from "tone";

/** sample
 * Purpose: takes a sound and a number for volume and returns a sample to play notes with. Used mainly in play-related function
 */

const sample = (sound, volume) => {
  const vol = new Volume(volume).toDestination();
  return new Sampler({
    urls: { C3: sound }
  }).connect(vol);
};

export default sample;