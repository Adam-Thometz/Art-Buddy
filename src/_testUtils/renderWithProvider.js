import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;