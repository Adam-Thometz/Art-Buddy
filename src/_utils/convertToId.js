export default function convertToId(name) {
  const nameArr = name.split(' ');
  let id = '';
  id += nameArr[0].toLowerCase();
  for (let i = 1; i < nameArr.length; i++) {
    nameArr[i].toLowerCase();
    nameArr[i][0].toUpperCase();
    id += nameArr[i];
  };
  return id;
};