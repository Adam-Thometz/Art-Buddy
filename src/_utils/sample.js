import { Sampler } from "tone";

const sample = (sound) => (
  new Sampler({
    urls: {
      C3: sound
    }
  }).toDestination()
);

export default sample;