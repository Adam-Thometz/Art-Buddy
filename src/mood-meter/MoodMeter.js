import React, { useContext, useEffect } from 'react';
import useVisited from '_hooks/visited/useVisited';
import { PopupContext } from '_context/PopupContext';

import { useDispatch, useSelector } from 'react-redux';
import { changeCurrGame } from '_redux/_general/generalActions';

import WindowNavbar from '_components/window-nav/WindowNavbar';
import Instructions from '_components/instructions/Instructions';
import Valence from './valence/Valence';
import Energy from './energy/Energy';

import { moodMeter } from '_data/_activities/activityList';
import Result from './result/Result';

const MoodMeter = () => {
  const [hasVisited] = useVisited(moodMeter.lsKey);
  const { valence, energy } = useSelector(state => state.moodMeter);
  const { currGame } = useSelector(state => state.general);
  const { setCurrPopup } = useContext(PopupContext)
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(changeCurrGame(moodMeter));
    return () => {
      dispatch(changeCurrGame({}));
    };
  }, [dispatch]);

  useEffect(() => {
    if (valence !== null && energy !== null) {
      setCurrPopup({
        title: 'YOUR MOOD IS...',
        popup: <Result />
      });
    };
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [valence, energy]);

  return (<>
    <WindowNavbar page={currGame.name} />
    {!hasVisited ? <Instructions /> : <>
      <Valence />
      <Energy />
    </>}
  </>);
};

export default MoodMeter;