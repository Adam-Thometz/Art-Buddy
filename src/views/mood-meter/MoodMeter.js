import { useContext, useEffect } from "react";
import useVisited from "hooks/visited/useVisited";
import { PopupContext } from "context/PopupContext";

import { useDispatch, useSelector } from "react-redux";
import { changeCurrGame } from "store/_general/generalReducer";

import WindowNavbar from "components/window-nav/WindowNavbar";
import Instructions from "components/instructions/Instructions";
import Valence from "./valence/Valence";
import Energy from "./energy/Energy";

import { moodMeter } from "data/_activities/activityList";
import Result from "./result/Result";

const MoodMeter = () => {
  const [hasVisited, setHasVisited] = useVisited(moodMeter.lsKey);
  const { valence, energy } = useSelector((state) => state.moodMeter);
  const { currGame } = useSelector((state) => state.general);
  const { setCurrPopup } = useContext(PopupContext);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(changeCurrGame(moodMeter));
    return () => {
      dispatch(changeCurrGame({}));
    };
  }, [dispatch]);

  useEffect(() => {
    if (valence !== null && energy !== null) {
      setCurrPopup({
        title: "YOUR MOOD IS...",
        popup: <Result />,
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [valence, energy]);

  return (
    <>
      <WindowNavbar page={currGame.name} />
      {!hasVisited ? (
        <Instructions setVisited={setHasVisited} />
      ) : (
        <>
          <Valence />
          <Energy />
        </>
      )}
    </>
  );
};

export default MoodMeter;
