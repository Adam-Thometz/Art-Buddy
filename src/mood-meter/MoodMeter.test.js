import React from "react";

import renderWithProvider from '_testUtils/renderWithProvider';

import MoodMeter from './MoodMeter';

import { MM } from '_data/_utils/localStorageKeys';

describe('MoodMeter component', () => {
  window.localStorage.setItem(`visited-${MM}`, true);
  it('renders without crashing', () => {
    renderWithProvider(<MoodMeter />);
  });
  
  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<MoodMeter />);
    expect(asFragment()).toMatchSnapshot();
  });
});