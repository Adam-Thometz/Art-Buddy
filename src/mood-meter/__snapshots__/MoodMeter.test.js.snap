// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodMeter component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="WindowNavbar-games"
  >
    <h1
      class="WindowNavbar-label"
    >
      MOOD METER
    </h1>
  </header>
  <section
    class="Valence"
  >
    <p>
      How are you feeling today?
    </p>
    <div
      class="Valence-faces"
    >
      <div
        class="Face"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
      </div>
      <div
        class="Face"
        id="1"
      >
        <img
          alt=""
          id="1"
          src="file"
        />
      </div>
      <div
        class="Face"
        id="0"
      >
        <img
          alt=""
          id="0"
          src="file"
        />
      </div>
    </div>
  </section>
  <section
    class="Energy"
  >
    <p>
      How is your energy level?
    </p>
    <div
      class="Energy-batteries"
    >
      <div
        class="Battery"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
        <img
          alt=""
          id="2"
          src="file"
          style="width: 89%;"
        />
      </div>
      <div
        class="Battery"
        id="1"
      >
        <img
          alt=""
          id="1"
          src="file"
        />
        <img
          alt=""
          id="1"
          src="file"
          style="width: 48%;"
        />
      </div>
      <div
        class="Battery"
        id="0"
      >
        <img
          alt=""
          id="0"
          src="file"
        />
        <img
          alt=""
          id="0"
          src="file"
          style="width: 12%;"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`MoodMeter component matches snapshot 2`] = `
<DocumentFragment>
  <header
    class="WindowNavbar-games"
  >
    <h1
      class="WindowNavbar-label"
    >
      MOOD METER
    </h1>
  </header>
  <section
    class="Valence"
  >
    <p>
      How are you feeling today?
    </p>
    <div
      class="Valence-faces"
    >
      <div
        class="Face"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
      </div>
      <div
        class="Face"
        id="1"
      >
        <img
          alt=""
          id="1"
          src="file"
        />
      </div>
      <div
        class="Face"
        id="0"
      >
        <img
          alt=""
          id="0"
          src="file"
        />
      </div>
    </div>
  </section>
  <section
    class="Energy"
  >
    <p>
      How is your energy level?
    </p>
    <div
      class="Energy-batteries"
    >
      <div
        class="Battery"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
        <img
          alt=""
          id="2"
          src="file"
          style="width: 89%;"
        />
      </div>
      <div
        class="Battery"
        id="1"
      >
        <img
          alt=""
          id="1"
          src="file"
        />
        <img
          alt=""
          id="1"
          src="file"
          style="width: 48%;"
        />
      </div>
      <div
        class="Battery"
        id="0"
      >
        <img
          alt=""
          id="0"
          src="file"
        />
        <img
          alt=""
          id="0"
          src="file"
          style="width: 12%;"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
